### PR TITLE
Make sure that Salad isn't mining when SaladBind is about to start

### PR DESCRIPTION
**Pull request by TheBlueBurger**
Only works on Windows and uses child processes since I can't figure another way out

### CODE REVIEW

File: src/mining.js [hunk 1]
```diff
@@ -462,6 +484,18 @@ async function prepStart(minerData, algo, pool, region, advancedCommands, quick=
 				"advancedCommands": advancedCommands
 			});
 			presence.mine(minerData.miner, algo, pool.name)
+			// Check if miner is already running in salad using child processes to run "tasklist" if on windows.
+			if(await saladMining) {
+				{
```
Comment (vtheskeleton) [inline src/mining.js:489]: did you do double brackets here lol

**Review by vtheskeleton — CHANGES_REQUESTED** [review]

File: src/mining.js [hunk 2]
```diff
@@ -431,6 +431,28 @@ async function selectPool(minerData, algo) {
 async function prepStart(minerData, algo, pool, region, advancedCommands, quick=false) {
 	if (advancedCommands == undefined) advancedCommands = ""
 	console.clear();
+	let saladMining = new Promise((resolve, reject) => {
+		if (process.platform == "win32") {
```
Comment (vtheskeleton) [inline src/mining.js:435]: Also also we use systeminformation for platform!! 😡

**Review by vtheskeleton — CHANGES_REQUESTED** [review]

File: src/mining.js (hunk 1 above)
Comment (TheBlueBurger) [reply src/mining.js:489]: You didn't see anything

File: src/mining.js (hunk 2 above)
Comment (TheBlueBurger) [reply src/mining.js:435]: Is this really a real issue, process.platform doesnt need to wait 10 years to finish and its way easier, you'll also get the same output (except the weird "Windows" issue)

Comment (vtheskeleton) [reply src/mining.js:435]: no i just noticed you did that lol

**Comment by vtheskeleton** [comment]
This is tested right?

File: src/mining.js [hunk 3]
```diff
@@ -431,6 +431,28 @@ async function selectPool(minerData, algo) {
...
+			try {
+
+				let processes = execSync("tasklist").toString();
+				const saladMiners = [
```
Comment (vtheskeleton) [inline src/mining.js:439]: Miner list seems a bit weird to me

**Comment by TheBlueBurger** [comment]
> 
> 
> This is tested right?

Tester and works for me at least

**Comment by vtheskeleton** [comment]
mkay

**Review by vtheskeleton — APPROVED** [review]
Will release yet another test build

**Pull request merged**